src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -263,7 +292,18 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
                     }
                 }
 
-                await ProcessSearchResultsAsync(searchResult.Results, currentBatchId, cancellationToken);
+                if (IsAnonymizedExportJob())
+                {
+                    string configurationWithEtag =
+                        $"{_exportJobRecord.AnonymizationConfigurationLocation}:{_exportJobRecord.AnonymizationConfigurationFileETag}";
+
+                    IAnonymizer anonymizer = await _anonymizerFactory.Value.CreateAnonymizerAsync(configurationWithEtag, cancellationToken);
```

Comment:
Could the anonymizer be created outside the while loop? That would avoid recreating it for every page of results.